src/Microsoft.IdentityModel.Tokens/Json/JsonSerializerPrimitives.cs

```diff
@@ -1129,6 +1129,10 @@ public static void WriteObject(ref Utf8JsonWriter writer, string key, object obj
 
```

Comment:
Not for this PR but this entire `if`-`else if`... chain I think should just be a giant `switch`. I'm also curious if the `IDictionary` handler works since the generic form doesn't inherit from the non-generic form.

Reply:
> I'm also curious if the IDictionary handler works since the generic form doesn't inherit from the non-generic form.

It does. `is IDictionary` similarly returns true for generic one.